sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/jdbc/JDBCRDD.scala

```diff
@@ -94,6 +96,74 @@ object JDBCRDD extends Logging {
     new StructType(columns.map(name => fieldMap(name)))
   }
 
+  def translateFilterV1ToV2(f: Filter): Option[Predicate] = {
```

Comment:
Instead of converting each of the V1 Filters here, can we define 
```
private[sql] def toV2: Predicate
```
in V1 `Filter`? In this way I can reuse the V1 to V2 code in file source. I will start migrating V1 Filter to V2 Filter in file source after this PR is merged.

Reply:
Sorry, I didn't make myself clear. What I meant is to move the `translateFilterV1ToV2` out of JDBC so I can reuse this in file source. In `sources.filters`, have something like this:
```
sealed abstract class Filter {
  ......
  private[sql] def toV2: Predicate
}
```
Then implement this in each of the V1 Filters, for example:
```
case class EqualTo(attribute: String, value: Any) extends Filter {
  ......
  def toV2: Predicate = {
    checkColumnName(attribute)
    new Predicate("=",
      Array(FieldReference(attribute), LiteralValue(value, Literal(value).dataType)))
  }
}
```

Reply:
OK